lib/services/publish-service.ts

```diff
@@ -0,0 +1,211 @@
+            this.$project.importProject().wait();
+            this.$logger.printInfoMessageOnSameLine("Publishing");
+
+            try {
+                this.$progressIndicator.showProgressIndicator(this.$server.publish.publishToFtp(projectName, projectName, ftpConnectionData), 6000).wait();
+            } catch (err) {
+                this.$logger.printInfoMessageOnSameLine(os.EOL);
+                if (err.message === PublishService.ERROR_MESSAGE_WHEN_PROJECT_EXISTS_ON_REMOTE) {
+                    this.$errors.failWithoutHelp("This project already exists on the remote server. " + os.EOL + 
+                                    "To remove all files and upload the new ones on the remote run --force option");
+                }
+
+                throw err;
+            }
+
+            this.$logger.printInfoMessageOnSameLine(os.EOL);
+            this.$logger.info("Project '%s' published successfully.", projectName);
+        }).future<void>()();
+    }
+
+    public listAllConnections(): IFuture<void> {
+        return (() => {
+            this.listConnections(this.ftpPublishConnections);
+
+            if (!this.ftpPublishConnections.length) {
+                this.$logger.info("No connections found. To add a connection, run `connection add`");
+            }
+        }).future<void>()();
+    }
+
+    public addConnection(name: string, publishUrl: string): IFuture<void> {
+        return (() => {
+            var ftpPublishConnection: IPublishConnection = {
+                type: PublishService.ALLOWED_COONECTION_TYPE,
+                publicUrl: PublishService.PUBLIC_URL_DEFAULT_VALUE,
+                publishUrl: publishUrl,
+                name: name
+            };
+
+            if (!ftpPublishConnection.name) {
+                ftpPublishConnection.name = this.$prompter.getString("Name:").wait();
+            }
+
+            if (!ftpPublishConnection.publishUrl) {
+                ftpPublishConnection.publishUrl = this.$prompter.getString("Publish URL:").wait();
+            }
+
+            this.allPublishConnections.push(ftpPublishConnection);
+            this.savePublishJsonFile();
+            this.$logger.info("Connection '%s' added successfully.", ftpPublishConnection.name);
+        }).future<void>()();
+    }
+
+    public removeConnection(idOrName: string): IFuture<void> {
+        return (() => {
+            var id = parseInt(idOrName);
+            var message : string;
+            if (!isNaN(id)) {
+                this.validateId(id);
+
+                // (id - 1) is the index of the object to be deleted
+                _.remove(this.allPublishConnections, this.ftpPublishConnections[id - 1]);
+                message = util.format("Connection '%s' deleted successfully", this.ftpPublishConnections[id - 1].name);
+            } else {
+                var name = idOrName;
+                var connectionsToBeRemoved = _.filter(this.allPublishConnections, function(connection) {
```

Comment:
`_.filter(this.allPublishConnections, (connection) => connection.name === name)` is better